### DISCUSSION TITLE
Continuous Integration

### DISCUSSION THREAD

**Discussion by kaaloo** [observablehq/framework#999] [Q&A]
How could the `observable deploy` command work in non interactive mode for running in a GitHub actions workflow?

**Comment by mbostock** [comment]
That should already be the case. We use [`isatty`](https://nodejs.org/docs/latest-v20.x/api/tty.html#ttyisattyfd) to check whether the stdin is interactive:

https://github.com/observablehq/framework/blob/92377f8e45290b2f69468d0a1c40a710a20efb63/src/tty.ts#L31

Are you seeing something else? Note that you need a `docs/.observablehq/deploy.json` to run deploy non-interactively.
